src/plugin.test.tsx

```diff
@@ -116,6 +116,34 @@ describe('argo-cd', () => {
       expect(await rendered.findByText('https://github.com/argoproj/argocd-example-apps')).toBeInTheDocument();
     });
 
+    it('should display new data on retry', async () => {
+      worker.use(
+        rest.get('*', (_, res, ctx) => res(ctx.json(getResponseStub)))
+      );
+
+      const rendered = render(
+        <ApiProvider apis={apis}>
+          <ArgoCDDetailsWidget entity={getEntityStub} />
+        </ApiProvider>
+      );
+
+      expect(await rendered.findByText('guestbook')).toBeInTheDocument();
+      expect(await rendered.findByText('Synced')).toBeInTheDocument();
+
+      const nextResponseStub = getResponseStub;
+      nextResponseStub.status.sync.status = "OutOfSync";
+
+      worker.use(
+        rest.get('*', (_, res, ctx) => res(ctx.json(getResponseStub)))
+      );
+
+      const refreshButton = await rendered.findByTitle("Refresh");
+      refreshButton.click();
```

Comment:
Something like 
```
fireEvent.click(refreshButton)
```
here, where fireEvent is from `testing-library/react`

Reply:
Thanks